zcash_client_sqlite/src/testing.rs

```diff
@@ -1099,24 +1152,22 @@ fn fake_compact_block_spending<P: consensus::Parameters, Fvk: TestFvk>(
     let mut rng = OsRng;
     let mut ctx = fake_compact_tx(&mut rng);
 
-    // Create a fake spend
-    fvk.add_compact_spend(&mut ctx, nf, &mut rng);
-
-    // Create a fake Note for the payment
-    ctx.outputs
-        .push(compact_sapling_output(params, height, to, value, fvk.sapling_ovk(), &mut rng).0);
-
-    // Create a fake Note for the change
-    fvk.add_compact_output(
+    // Create a fake spend and a fake Note for the change
+    fvk.add_compact_action(
```

Comment:
Hm, this seems too restrictive; isn't `add_compact_action` only available when the `orchard` feature is enabled? I think that the change should go to whatever pool the output is targeting.

Reply:
This is the trait method, not the feature gated helper. I will name this `add_logical_action` to clarify.